include/libcyphal/transport/can/transport.hpp

```diff
@@ -27,55 +28,81 @@ namespace can
 class ICanTransport : public ITransport
 {};
 
+/// Internal implementation details of the CAN transport.
+/// Not supposed to be used directly by the users of the library.
+///
 namespace detail
 {
+
 class TransportImpl final : public ICanTransport, private TransportDelegate
 {
-    // In use to disable public construction.
-    // See https://seanmiddleditch.github.io/enabling-make-unique-with-private-constructors/
-    struct Tag
+    /// @brief Defines specification for making interface unique ptr.
+    ///
+    struct Spec
     {
-        explicit Tag()  = default;
         using Interface = ICanTransport;
         using Concrete  = TransportImpl;
     };
 
+    /// @brief Internal (private) storage of a media index, its interface and TX queue.
+    ///
+    /// B/c it's private, it also disables public construction of `TransportImpl`.
+    /// See https://seanmiddleditch.github.io/enabling-make-unique-with-private-constructors/
+    ///
+    class Media final
+    {
+    public:
+        Media(const std::size_t _index, IMedia& _interface, const std::size_t tx_capacity)
```

Comment:
Probably. I don't have yet enough expertise on this matter, but your logic looks right for me.

Reply:
>The tx_capacity is how big the TX queue will be in libcanard, correct? So, if I have a max message size of 2000 bytes and a CAN_FD MTU of 64, then my tx_capacity needs to be at least 32 frames?

Yes.

>We'll probably want to provide some guidance on setting this value in the user documentation when we get to that point.

Would it be appropriate to simply refer the reader to the lizard docs?

Reply:
Yes, I think we could add a comment in the Libcyphal code that points the user to the lizard documentation. I assume you're referring to this comment in canard: https://github.com/OpenCyphal/libcanard/blob/master/libcanard/canard.h#L244-L247

I think it might still be helpful to explain how to compute the maximum number of frames you will need, i.e. `MaxMessageSize / MTU`, even though this might be obvious. Maybe that's somewhere in the lizard docs?